packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -355,7 +356,11 @@ function hydrateChildren(
     node: Node | null,
     children: VNodes,
     parentNode: Element | ShadowRoot,
-    owner: VM
+    owner: VM,
+    // When rendering the children of a VFragment, additional siblings may follow the
+    // last node of the fragment. Hydration should not fail if a trailing sibling is
+    // found in this case.
+    expectAddlSiblings: boolean
```

Comment:
Yes. I'm trying to think of scenarios where there'd also be trailing nodes that are not comment or text nodes. Maybe if there's an adjacent fragment, but I'm not sure what combination of LWC template syntax would result in something like that.

I also considered making this a number, where the function would know the exact number of expected trailing nodes. But that would require piping the number throughout the hydration logic, and I couldn't figure a way to do it that wasn't messy.

Reply:
The following would result in an adjacent fragment:
```html
<div lwc:if={condition}>...</div>
<div lwc:if={condition}>...</div>
```

Reply:
That's a good example. It would be particularly difficult to know how many trailing nodes are expected, and I'm not sure what it buys us exactly. It seems like it is enough for this hydration function to know that trailing nodes are expected. If there's a mismatch in type for those trailing nodes, that'll be caught when `hydrateFragment` returns the last element corresponding to its children, and will be used to inspect the next node in the parent invocation of `hydrateChildren`.